corehq/util/couchdb_management.py

```diff
@@ -0,0 +1,63 @@
+from couchdbkit.client import Database
+from corehq.util.couch import get_document_class_by_doc_type
+from dimagi.utils.decorators.memoized import memoized
+from django.conf import settings
+
+
+class CouchConfig(object):
+    """
+    Interface for accessing the couch-related settings
+
+    You can use db_uri pass in a different couch instance
+    from the one specified by COUCH_DATABASE
+
+    """
+    def __init__(self, db_uri=None):
+        if db_uri:
+            self._settings_helper = (
+                settings.COUCH_SETTINGS_HELPER._replace(couch_database_url=db_uri))
+        else:
+            self._settings_helper = settings.COUCH_SETTINGS_HELPER
+
+    @property
+    def db_uri(self):
+        return self._settings_helper.couch_database_url
+
+    @property
+    @memoized
+    def all_db_uris_by_slug(self):
+        dbs = self._settings_helper.get_extra_couchdbs()
+        dbs[None] = self.db_uri
```

Comment:
@TylerSheffels asked what this line does. I guess it's not all that intuitive! `None` is the "slug" for the main db, so this line just adds the main db to the mapping. (`get_extra_couchdbs` only gets the "extra" i.e. non-main dbs.)